apiserver/facades/client/cloud/cloud_test.go

```diff
@@ -1376,267 +1891,27 @@ func (s *cloudSuite) TestCredentialContentsAllNoSecrets(c *gc.C) {
-func (st *mockBackend) AllCloudCredentials(user names.UserTag) ([]state.Credential, error) {
-	st.MethodCall(st, "AllCloudCredentials", user)
-	var result []state.Credential
-	for _, v := range st.creds {
-		result = append(result, v)
-	}
-	return result, st.NextErr()
-}
-
-func (st *mockBackend) CredentialModelsAndOwnerAccess(tag names.CloudCredentialTag) ([]state.CredentialOwnerModelAccess, error) {
-	st.MethodCall(st, "CredentialModelsAndOwnerAccess", tag)
-	return st.credsModels, st.NextErr()
-}
-
-func (st *mockBackend) CredentialModels(tag names.CloudCredentialTag) (map[string]string, error) {
-	st.MethodCall(st, "CredentialModels", tag)
-	return st.credentialModelsF(tag)
-}
-
-func (st *mockBackend) GetCloudAccess(cloud string, user names.UserTag) (permission.Access, error) {
-	st.MethodCall(st, "GetCloudAccess", cloud, user)
-	if cloud == "your-cloud" {
-		return permission.NoAccess, errors.NotFoundf("cloud your-cloud")
-	}
-	return st.cloudAccess, nil
-}
-
-func (st *mockBackend) GetCloudUsers(cloud string) (map[string]permission.Access, error) {
-	st.MethodCall(st, "GetCloudUsers", cloud)
-	return map[string]permission.Access{
-		"fred": permission.AddModelAccess,
-		"mary": permission.AdminAccess,
-	}, nil
-}
-
-func (st *mockBackend) CloudsForUser(user names.UserTag, all bool) ([]state.CloudInfo, error) {
-	st.MethodCall(st, "CloudsForUser", user, all)
-	return []state.CloudInfo{
-		{
-			Cloud:  st.cloud,
-			Access: permission.AddModelAccess,
-		},
-	}, nil
-}
-
-func (st *mockBackend) User(tag names.UserTag) (cloudfacade.User, error) {
-	st.MethodCall(st, "User", tag)
-	return &mockUser{tag.Name()}, nil
-}
-
-func (st *mockBackend) CreateCloudAccess(cloud string, user names.UserTag, access permission.Access) error {
-	st.MethodCall(st, "CreateCloudAccess", cloud, user, access)
-	if st.cloudAccess != permission.NoAccess {
-		return errors.AlreadyExistsf("access %s", access)
-	}
-	st.cloudAccess = access
-	return nil
-}
-
-func (st *mockBackend) UpdateCloudAccess(cloud string, user names.UserTag, access permission.Access) error {
-	st.MethodCall(st, "UpdateCloudAccess", cloud, user, access)
-	st.cloudAccess = access
-	return nil
-}
-
-func (st *mockBackend) RemoveCloudAccess(cloud string, user names.UserTag) error {
-	st.MethodCall(st, "RemoveCloudAccess", cloud, user)
-	st.cloudAccess = permission.NoAccess
-	return nil
-}
-
-func (st *mockBackend) RemoveModelsCredential(tag names.CloudCredentialTag) error {
-	st.MethodCall(st, "RemoveModelsCredential", tag)
-	return st.NextErr()
-}
-
-type mockUser struct {
-	name string
-}
-
-func (m *mockUser) DisplayName() string {
-	return "display-" + m.name
+func cloudCredentialTag(name, owner, cloudName string, attrs map[string]string) state.Credential {
+	cred := statetesting.CloudCredential(jujucloud.EmptyAuthType, attrs)
+	cred.Name = name
+	cred.Owner = owner
+	cred.Cloud = cloudName
+	return cred
 }
 
 type mockModel struct {
+	credentialcommon.PersistentBackend
+	credentialcommon.Model
```

Comment:
Why?

Reply:
This  make `mockModel` satisfy the embedded interfaces.